src/docker/shared.rs

```diff
@@ -241,37 +241,27 @@ impl DockerPaths {
     }
 
     pub fn mount_cwd(&self) -> &str {
-        &self.directories.mount_cwd
+        self.directories.mount_cwd()
     }
 
     pub fn host_root(&self) -> &Path {
-        &self.directories.host_root
+        self.directories.host_root()
     }
 }
 
 #[derive(Debug)]
-pub struct Directories {
-    pub cargo: PathBuf,
-    pub xargo: PathBuf,
-    pub target: PathBuf,
-    pub nix_store: Option<PathBuf>,
-    pub host_root: PathBuf,
-    // both mount fields are WSL paths on windows: they already are POSIX paths
-    pub mount_root: String,
-    pub mount_cwd: String,
-    pub toolchain: QualifiedToolchain,
-    pub cargo_mount_path: String,
-    pub xargo_mount_path: String,
-    pub sysroot_mount_path: String,
+pub struct ToolchainDirectories {
+    cargo: PathBuf,
```

Comment:
I've intentionally made all the fields private to ensure we're using the functions, which allow us to have the same API for `Directories` and `ToolchainDirectories`, rather than have to do something like `&dirs.toolchain.cargo` for some, and `&dirs.cargo` for others.